libraries/ms-drivers/test/test_bts_7200_current_sense.c

```diff
@@ -285,40 +394,516 @@ void test_bts_7200_current_sense_stop_return_behaviour(void) {
   uint32_t interval_us = 500;            // 0.5 ms
   void *context_pointer = &interval_us;  // arbitrary pointer
   Bts7200Stm32Settings settings = {
     .select_pin = &test_select_pin,
     .sense_pin = &test_sense_pin,
+    .enable_0_pin = &test_enable_0_pin,
+    .enable_1_pin = &test_enable_1_pin,
     .interval_us = interval_us,
+    .resistor = BTS7200_TEST_RESISTOR,
     .callback = &prv_callback_increment,
     .callback_context = context_pointer,
   };
-  Bts7200Storage storage = { 0 };
 
-  TEST_ASSERT_OK(bts_7200_init_stm32(&storage, &settings));
-  TEST_ASSERT_OK(bts_7200_start(&storage));
-  TEST_ASSERT_EQUAL(true, bts_7200_stop(&storage));
+  TEST_ASSERT_OK(bts_7200_init_stm32(&s_storage, &settings));
+  TEST_ASSERT_OK(bts_7200_start(&s_storage));
+  TEST_ASSERT_EQUAL(true, bts_7200_stop(&s_storage));
   TEST_ASSERT_EQUAL(received_context, context_pointer);
 }
+
+// Test enabling/disabling/getting value works with IN0 pin
+void test_bts_7200_output_0_functions_work(void) {
+  // these don't matter (adc isn't reading anything) but can't be null
+  GpioAddress test_select_pin = { .port = GPIO_PORT_A, .pin = 0 };
+  GpioAddress test_sense_pin = { .port = GPIO_PORT_A, .pin = 0 };
+  // EN0 and EN1 pins
+  GpioAddress test_enable_0_pin = { .port = GPIO_PORT_A, .pin = 1 };
+  GpioAddress test_enable_1_pin = { .port = GPIO_PORT_A, .pin = 2 };
+  uint32_t interval_us = 500;  // 0.5 ms
+  Bts7200Stm32Settings settings = {
+    .select_pin = &test_select_pin,
+    .sense_pin = &test_sense_pin,
+    .enable_0_pin = &test_enable_0_pin,
+    .enable_1_pin = &test_enable_1_pin,
+    .interval_us = interval_us,
+    .resistor = BTS7200_TEST_RESISTOR,
+    .callback = &prv_callback_increment,
+  };
+  TEST_ASSERT_OK(bts_7200_init_stm32(&s_storage, &settings));
+
+  // Should initialize open
+  TEST_ASSERT_EQUAL(bts_7200_get_output_0_enabled(&s_storage), false);
+
+  // Close, then check
+  TEST_ASSERT_OK(bts_7200_enable_output_0(&s_storage));
+  TEST_ASSERT_EQUAL(bts_7200_get_output_0_enabled(&s_storage), true);
+
+  // Open, then check
+  TEST_ASSERT_OK(bts_7200_disable_output_0(&s_storage));
+  TEST_ASSERT_EQUAL(bts_7200_get_output_0_enabled(&s_storage), false);
+
+  // Make sure possible to open while already open
+  TEST_ASSERT_OK(bts_7200_disable_output_0(&s_storage));
+  TEST_ASSERT_EQUAL(bts_7200_get_output_0_enabled(&s_storage), false);
+
+  // Make sure starting current sense doesn't cause issues
+  TEST_ASSERT_OK(bts_7200_start(&s_storage));
+
+  // Same tests as above:
+  TEST_ASSERT_EQUAL(bts_7200_get_output_0_enabled(&s_storage), false);
+  TEST_ASSERT_OK(bts_7200_enable_output_0(&s_storage));
+  TEST_ASSERT_EQUAL(bts_7200_get_output_0_enabled(&s_storage), true);
+  TEST_ASSERT_OK(bts_7200_disable_output_0(&s_storage));
+  TEST_ASSERT_EQUAL(bts_7200_get_output_0_enabled(&s_storage), false);
+  TEST_ASSERT_OK(bts_7200_disable_output_0(&s_storage));
+  TEST_ASSERT_EQUAL(bts_7200_get_output_0_enabled(&s_storage), false);
```

Comment:
The order for the `TEST_ASSERT_` macros is `expected, actual`, so true/false should be first. However there are also `TEST_ASSERT_TRUE` and `TEST_ASSERT_FALSE` macros which really should be used instead (even though I realize there's a bunch of `TEST_ASSERT_EQUAL(true, ...)` in the file already)